### ISSUE TITLE
Connection closed before full header was received

### ISSUE THREAD

**Issue by chengqing97** [dart-lang/sdk#48450]
## Issue:
I am getting this error when sending requests to some URLs:
```
Unhandled exception:
HttpException: Connection closed before full header was received, uri = https://dictionary.cambridge.org/dictionary/english/hello
```
## My environment:
Dart: Dart SDK version: 2.16.1 (stable) on "linux_x64"
OS: Debian 11 bullseye
Kernel: x86_64 Linux 5.10.0-11-amd64

## Steps to reproduce:
1. Create a `main.dart`
2. Inside `main.dart`:
```
import 'dart:convert';
import 'dart:io';

void main(List<String> arguments) async {
  final url1 = Uri.parse("https://dictionary.cambridge.org/dictionary/english/hello");
  final url2 = Uri.parse("https://dict.youdao.com/w/hello");

  var client = HttpClient();
  try {
    HttpClientRequest request = await client.getUrl(url1);
    HttpClientResponse response = await request.close();
    final stringData = await response.transform(utf8.decoder).join();
    print(stringData);
  } finally {
    client.close();
  }
}

```

3. Run the code `dart run main.dart`

I can make request to `url2` just fine. I'm only getting error when making request to `url1`

**Comment by SuporteSysrs01** [comment]
Probably a duplicate of flutter/flutter/#41573

**Comment by brianquinlan** [comment]
It looks like "dictionary.cambridge.org" is doing some sort of user agent sniffing. 

If I change the user agent with:
```
client.userAgent = 'curl/7.77.0';
```

Then the server produces a valid HTTP response.

**Comment by Rishyash** [comment]
For those who are getting these crash as fatal crash.
https://medium.com/@rishyash8/flutter-firebase-crash-report-because-of-connection-closed-before-full-header-was-received-7ace6e84bc5d
